packages/plugin-ext/src/main/browser/webviews-main.ts

```diff
@@ -25,24 +25,47 @@ import { WebviewWidget } from './webview/webview';
 import { ThemeService } from '@theia/core/lib/browser/theming';
 import { ThemeRulesService } from './webview/theme-rules-service';
 import { DisposableCollection } from '@theia/core';
+import { ViewColumnService } from './view-column-service';
 
 export class WebviewsMainImpl implements WebviewsMain {
+    private readonly revivers = new Set<string>();
     private readonly proxy: WebviewsExt;
     protected readonly shell: ApplicationShell;
+    protected readonly viewColumnService: ViewColumnService;
     protected readonly keybindingRegistry: KeybindingRegistry;
     protected readonly themeService = ThemeService.get();
     protected readonly themeRulesService = ThemeRulesService.get();
 
     private readonly views = new Map<string, WebviewWidget>();
+    private readonly viewsOptions = new Map<string, { panelOptions: WebviewPanelShowOptions; panelId: string; active: boolean; visible: boolean; }>();
 
     constructor(rpc: RPCProtocol, container: interfaces.Container) {
         this.proxy = rpc.getProxy(MAIN_RPC_CONTEXT.WEBVIEWS_EXT);
         this.shell = container.get(ApplicationShell);
         this.keybindingRegistry = container.get(KeybindingRegistry);
+        this.viewColumnService = container.get(ViewColumnService);
+
+        let timeoutHandle: NodeJS.Timer | undefined;
+        const updateOptions = () => {
+            if (!this.viewsOptions.size) {
+                return;
+            }
+            if (timeoutHandle) {
+                clearTimeout(timeoutHandle);
+            }
+            timeoutHandle = setTimeout(() => {
```

Comment:
Is it because view is not rendered yet?

Reply:
I am really sorry. I accidentally pushed my last changes with force. I forgot about review status.

Reply:
Yes. It is because the view is not rendered yet.  I do not know what can be used instead.

Reply:
@akosyakov
Do you know what can be used instead?

Reply:
I have one idea. I am checking

Reply:
I can't find anything that can be used instead of a timer.  Without this timer, my event can be called three times instead of one.

@akosyakov WDYT? Could we live this timer as is?

Reply:
ok, so you are just debouncing it, fyi you could use `lodash-debounce` instead timers

Reply:
fixed